src/poly/commitment.rs

```diff
@@ -21,6 +23,95 @@ pub struct OpeningProof<C: CurveAffine> {
     z2: C::Scalar,
 }
 
+/// A multiscalar multiplication in the polynomial commitment scheme
+#[derive(Debug)]
+pub struct MSM<C: CurveAffine> {
+    /// TODO: documentation
+    pub g_scalars: Option<Vec<C::Scalar>>,
+
+    /// TODO: documentation
+    pub h_scalar: Option<C::Scalar>,
+
+    /// TODO: documentation
+    pub other_scalars: Vec<C::Scalar>,
+
+    /// TODO: documentation
+    pub other_bases: Vec<C>,
+}
+
+impl<'a, C: CurveAffine> MSM<C> {
+    /// Empty MSM
+    pub fn default(params: &Params<C>) -> Self {
+        let g_scalars = Some(vec![C::Scalar::one(); params.n as usize]);
```

Comment:
As mentioned, this should be a method of `Params`, but also we shouldn't be initializing `g_scalars` and `h_scalar` to anything. We can set them to both to `None` that way if we call the `MSM` and there are no scalars, it doesn't try to do a multiexp using the `g` and `h` bases from params, since it doesn't need to.